### DISCUSSION TITLE
use array as state rather than object

### DISCUSSION THREAD

**Discussion by sanjeethboddi** [pmndrs/zustand#2662] [Q&A]
I want to use array instead of object. Would this work?
```
const useStore = create((set) => ([
  1,
  () => set((state) => (state[0] + 1 )),
]))

function Counter() {
  const { count, inc } = useStore((state)=>state[0])
  return (
    <div class="counter">
      <span>{count}</span>
      <button onClick={inc}>one up</button>
    </div>
  )
}
```

**Comment by dbritto-dev** [comment]
@sanjeethboddi here you go -> https://github.com/pmndrs/zustand/blob/ba484ceab003072205493670b94ab83842f7a5a2/docs/reference/create.md#updating-arrays-in-state

**Comment by dai-shi** [comment]
I just tweeted about something similar yesterday: https://x.com/dai_shi/status/1819363657324912660
